runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -366,6 +366,9 @@ TR_J9InlinerPolicy::alwaysWorthInlining(TR_ResolvedMethod * calleeMethod, TR::No
       case TR::java_lang_Class_newInstance:
       case TR::jdk_internal_util_Preconditions_checkIndex:
 
+      // AbstractMemorySegmentImpl.reinterpret methods call Reflection.getCallerClass
+      case TR::jdk_internal_foreign_AbstractMemorySegmentImpl_reinterpret:
```

Comment:
Does this only need to be done for JDK22+?  If so, this case should be guarded with a JDK level check.

Reply:
Actually, I don't think we usually add conditional compilation for methods that exist in some versions of the Java Class Libraries but not others.  Unless there's really a difference in the behaviour for different versions &mdash; as opposed to a difference in whether it exists &mdash; I don't think this needs to be conditionally compiled in or out.

Reply:
That was the reasoning behind my question: whether there was a difference in behaviour between 21 and 22+ and if you only need to affect behaviour this way for JDK22+.  If the answer is no (and I just now compared the code myself and see that they are largely similar) then guarding it for a JDK level is not necessary.

Reply:
@0xdaryl I misunderstood your question and added the guard. I have now removed that change.